outdated-pkgs

```diff
@@ -0,0 +1,305 @@
+
+def get_osg_list():
+    if SERIES == 'devops':
+        osg_tag = "%s-el%s-%s" % (SERIES, EL, TAG)
+    else:
+        osg_tag = "osg-%s-el%s-%s" % (SERIES, EL, TAG)
+    cmd = "osg-koji list-tagged --latest --rpms %s" % osg_tag
+    cache_name = osg_tag + '.list'
+
+    handle = get_handle(os.popen, cmd, cache_name)
+
+    nvrs = [ line[:-5] for line in handle if line.endswith(".src\n") ]
+    return strip_dist_tag(nvrs, 'osg')
+
+def rpmvercmp(a,b):
+    return rpm.labelCompare(*[stringToVersion(x) for x in (a,b)])
+
+def n_vr(nvr):
+    n,v,r = nvr.rsplit("-",2)
+    return n, v+'-'+r
+
+def nvrmap(seq):
+    return dict( n_vr(nvr) for nvr in seq )
+
+if make_cache:
+    if MODE != 'goc':
+        # osg scripts both call this function
+        get_osg_list()
+    if MODE == 'osg-epel':
+        get_epel_list()
+    if MODE == 'osg-i2':
+        get_i2_list()
+    sys.exit()
+
+def colorize(color, *seq):
+    return [ "\x1b[%sm%s\x1b[0m" % (color, x) for x in seq ]
+
+def colorize_vr(vr1, vr2):
+    v1,r1 = vr1.split('-')
+    v2,r2 = vr2.split('-')
+
+    if v1 != v2:
+        v1,v2 = colorize('1;32', v1, v2)
+    elif r1 != r2:
+        r1,r2 = colorize('1;34', r1, r2)
+
+    return map('-'.join, [[v1,r1],[v2,r2]])
+
+def colorize_html(color, *seq):
+    return [ '<span class="%s">%s</span>' % (color, x) for x in seq ]
+
+def colorize_html_vr(vr1, vr2):
+    v1,r1 = vr1.split('-')
+    v2,r2 = vr2.split('-')
+
+    if v1 != v2:
+        v1,v2 = colorize_html('vdiff', v1, v2)
+    elif r1 != r2:
+        r1,r2 = colorize_html('rdiff', r1, r2)
+
+    return map('-'.join, [[v1,r1],[v2,r2]])
+
+
+
+if html_out:
+    colorize_vr = colorize_html_vr
+    print ("<html>\n<head>\n<style type='text/css'>\n.vdiff {color:green}\n"
+           ".rdiff {color:blue}\n</style>\n</head>\n<body>\n<pre>")
+
+if preamble:
+    print preamble
+    print
+
+pkg_diffs = []
+
+def get_pkg_diff(source, compare):
+     for pkg in sorted(source):
+        if pkg in compare:
+            vrcmp = rpmvercmp(source[pkg], compare[pkg])
+            if vrcmp < 0 or show_all:
+                pkg_diffs.append([pkg, source[pkg], compare[pkg]])
+
+def handle_pkg_diff():
+    if pkg_diffs:
+        if MODE == 'goc':
+            pkg_diffs[:0] = [["Package", "EL%s GOC" % EL, "OSG " + SERIES]]
+        elif MODE == 'osg-epel':
+            pkg_diffs[:0] = [["Package", "OSG " + SERIES, "EPEL " + EL]]
+        elif MODE == 'osg-i2':
+            pkg_diffs[:0] = [["Package", "OSG " + SERIES, "I2 EL" + EL]]
+        widths = [ max(map(len,col)) for col in zip(*pkg_diffs) ]
+        pkg_diffs[1:1] = [[ '-' * n for n in widths ]]
+        for i,row in enumerate(pkg_diffs):
+            spacing = [ w-len(x) for x,w in zip(row,widths) ]
+            if use_color and i > 1:
+                row[1:] = colorize_vr(*row[1:])
+            print '  '.join( r + ' ' * s for r,s in zip(row,spacing) ).rstrip()
+    else:
+        print "No package version differences"
```

Comment:
This is a good start for breaking out the common functionality.

I like your idea of making separate functions for `get_pkg_diff()` and `handle_pkg_diff()` - since each block of code has its own discrete logical task.  But if you are going to separate them this way, i would suggest that `pkg_diffs` can be returned by `get_pkg_diff()` and taken as a parameter to `handle_pkg_diff()`, rather than making it a global.  (Though it should work as you have it.)

Reply:
I guess you mean move the handle function under the other?

Reply:
Nevermind I get what you mean. I was trying to keep the original logic since I'm not sure if there are any special considerations behind setting that list to global.